src/core/ddsi/src/ddsi__cfgelems.h

```diff
@@ -103,6 +103,21 @@ static struct cfgelem psmx_attributes[] = {
       "This has no meaning in CycloneDDS itself, and its parsing is deferred to the"
       "PSMX implementation.</p>"
     )),
+  STRING("forbiddenTopics", NULL, 1, "",
+    MEMBEROF(ddsi_config_psmx_listelem, cfg.forbidden_topics),
+    FUNCTIONS(if_topic_array, uf_topic_array, ff_topic_array, pf_topic_array),
```

Comment:
I would go for exactly the same pattern as what is used in the "network partitions" (and the "ignored partitions", so https://github.com/eclipse-cyclonedds/cyclonedds/blob/00c63c342c2d858bf94b6a602dcf8bd696fdbe26/src/core/ddsi/src/ddsi__cfgelems.h#L739). Not because it is necessarily better, but simply because it means more code can be reused.

Doing it using multiple XML elements also means you don't have to worry (as much) about what characters are allowed in topic names 😀

Reply:
@eboasson Thank you! I had the feeling that this problem must have come up in some other option already, I didn't think of the "ignored partitions" setting.

Reply:
@eboasson Moved the patterns into separate XML elements. I had some trouble figuring out how to work with XML text content, so the config now looks something like this

```
<PubSubMessageExchange name="iox" library="psmx_iox" config="LOG_LEVEL=INFO;">
  <forbiddenTopics>
    <Pattern value="Round?rip"/> <!-- note that the value is an attribute, not element content -->
    <Pattern value="*Hello*"/>
  </forbiddenTopics>
</PubSubMessageExchange>
```

If you'd prefer the pattern to be the text content of the `Pattern` element I'd need some pointers on how to do that :-).